modules/caddyhttp/caddyauth/command.go

```diff
@@ -52,16 +60,50 @@ be provided (scrypt).
 }
 
 func cmdHashPassword(fs caddycmd.Flags) (int, error) {
+	var err error
+
 	algorithm := fs.String("algorithm")
 	plaintext := []byte(fs.String("plaintext"))
 	salt := []byte(fs.String("salt"))
 
 	if len(plaintext) == 0 {
-		return caddy.ExitCodeFailedStartup, fmt.Errorf("password is required")
+		if terminal.IsTerminal(int(os.Stdin.Fd())) {
+			fmt.Print("Enter password: ")
+			plaintext, err = terminal.ReadPassword(int(os.Stdin.Fd()))
+			fmt.Println()
+
+			if err != nil {
+				return caddy.ExitCodeFailedStartup, err
+			}
```

Comment:
Error check should come immediately after the line that produces the error, for better readability and maintainability (same below).

Unless, does the terminal not go to the next line _at all_ without the `Println()`? Or is the Println adding an _extra_ (empty) line?

Reply:
I tested it, and ReadPassword does not preserve the newline.